strawberry/types/federation/object_type.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Optional, Union
+
+from strawberry.types import StrawberryObjectType, StrawberryUnion
+
+_OBJECT_FED_TYPE = Union[StrawberryObjectType, 'StrawberryObjectTypeFederation']
+
+
+# TODO: Should federation use composition or inheritance?
+#       i.e. should StrawberryObjectTypeFederation inherit from
+#       StrawberryObjectType or just hold a reference to one?
+
+# TODO: This name is probably too long and too verbose. Think of something
+class StrawberryObjectTypeFederation:
+    """
+
+    TODO: One decorator for federation? Or multiple? This feels a bit stacked
+    >>> @strawberry.key("upc")
+    ... @strawberry.extends(OtherType)
+    ... @strawberry.type
+    ... class ThisType:
+    ...     @strawberry.field
+    ...     def resolver(self) -> OtherType:
+
+    TODO: This is probably better?
+    >>> @strawberry.federation(extends=OtherType, key="upc")
```

Comment:
I prefer `strawberry.federation.field` :D

Reply:
but this is around a type, not a field?

Reply:
right, but we also have both :)

`strawberry.federation.field` and `strawberry.federation.type`